src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -299,7 +301,7 @@ MultiPlanTree(Query *queryTree)
 		 * elements.
 		 */
 		joinClauseList = JoinClauseList(whereClauseList);
-		tableEntryList = TableEntryList(rangeTableList);
+		tableEntryList = UsedTableEntryList(queryTree);
```

Comment:
They were being used for a different purpose. TableEntryList is called only from RelationIdList which is interested in all relation Ids used inside a query tree. This change is however deals with creating a join node, and any range table entry down below the subquery is irrelevant. 

I agree with you on not reporting unused range table entries in TableEntryList function too. I will check this again.